core/src/main/scala/com/mesosphere/usi/core/matching/OfferMatcher.scala

```diff
@@ -42,14 +43,23 @@ class OfferMatcher(masterDomainInfo: Mesos.DomainInfo) {
       case Nil =>
         result
 
-      case podSpec :: rest if !(podSpec.domainFilter(masterDomainInfo, originalOffer.getDomain)) =>
-        matchPodSpecsTaskRecords(originalOffer, remainingResources, result, rest)
-
-      case podSpec :: rest if !(podSpec.agentFilter(originalOffer)) =>
-        matchPodSpecsTaskRecords(originalOffer, remainingResources, result, rest)
-
       case podSpec :: rest =>
-        maybeMatchResourceRequirements(remainingResources, Nil, podSpec.runSpec.allResourceRequirements) match {
+        Some(podSpec).filter { podSpec =>
```

Comment:
This feels more terse and less readable than the previous variant.

Reply:
This is a fair point